components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/artifact/SAMLSSOArtifactResolutionService.java

```diff
@@ -0,0 +1,196 @@
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authenticator.samlsso.artifact;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.joda.time.DateTime;
+import org.opensaml.Configuration;
+import org.opensaml.common.SAMLObjectBuilder;
+import org.opensaml.common.SAMLVersion;
+import org.opensaml.saml2.core.Artifact;
+import org.opensaml.saml2.core.ArtifactResolve;
+import org.opensaml.saml2.core.ArtifactResponse;
+import org.opensaml.saml2.core.Issuer;
+import org.opensaml.saml2.core.RequestAbstractType;
+import org.opensaml.ws.soap.soap11.Envelope;
+import org.opensaml.xml.XMLObjectBuilderFactory;
+import org.wso2.carbon.identity.application.authenticator.samlsso.exception.ArtifactResolutionException;
+import org.wso2.carbon.identity.application.authenticator.samlsso.exception.SAMLSSOException;
+import org.wso2.carbon.identity.application.authenticator.samlsso.manager.X509CredentialImpl;
+import org.wso2.carbon.identity.application.authenticator.samlsso.util.SSOUtils;
+
+import java.util.Map;
+import java.util.UUID;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This class is used for handling SAML Artifact Binding
+ */
+public class SAMLSSOArtifactResolutionService {
+
+    private static Log log = LogFactory.getLog(SAMLSSOArtifactResolutionService.class);
+
+    private Map<String, String> authenticatorProperties;
+    private String tenantDomain;
+
+    public SAMLSSOArtifactResolutionService(Map<String, String> authenticatorProperties, String tenantDomain) {
+        this.authenticatorProperties = authenticatorProperties;
+        this.tenantDomain = tenantDomain;
+    }
+
+    /**
+     * Create a SAML artifactResolveObject based on given samlArt parameter and wrapped into a soapRequest
+     * Send the soapRequest to the Artifact Resolve Endpoint
+     *
+     * @param samlArt SAML Artifact reference needed to get the actual data
+     * @return ArtifactResponse
+     */
+    public ArtifactResponse getSAMLArtifactResponse(String samlArt) throws ArtifactResolutionException {
+
+        validateArtifactResolveConfig();
+        ArtifactResolve artifactResolve = generateArtifactResolveReq(samlArt);
+        return sendArtifactResolveRequest(artifactResolve);
+    }
+
+    /**
+     * Create SAML ArtifactResolve Object and sign
+     *
+     * @param sReceivedArtifact object reference to actual data
+     * @return SAML ArtifactResolve Object
+     */
+    public ArtifactResolve generateArtifactResolveReq(String sReceivedArtifact) throws ArtifactResolutionException {
+
+        ArtifactResolve artifactResolve = createArtifactResolveObject(sReceivedArtifact);
+        if (SSOUtils.isArtifactResolveReqSigned(authenticatorProperties)) {
+            if(log.isDebugEnabled()) {
+                log.debug("Signing artifact resolve request.");
```

Comment:
Shall we add the SAML art received to the debug log indicating signing the artifact resolve request for saml art received